[ADD] CRM: contact form leads (#9318)
odoo/documentation: content/applications/sales/crm/acquire_leads/opportunities_form.rst

========================================
Create opportunities from a contact form
========================================

Adding a contact form to a website makes it easy to convert visitors into leads and opportunities.
After a visitor submits their information, an opportunity can be created automatically, and assigned
to a designated sales team.

.. _crm/customize-contact-form:

Customize contact forms
=======================

By default, the *Contact Us* page on an Odoo website displays a preconfigured contact form. This
form can be customized, as needed, with additional fields and settings to suit the needs of a

> specific sales team.

Author: I toyed with the idea of listing out some examples (eg- "Adding a number field and asking approximately how many employees a company has" or "providing the space for customers to enter their budget") but I was concerned that adding it before all of the information would draw attention away from the actual necessary/required info. I'd rather have them focus on the details they need instead of giving them examples of things they *could* do

> Navigate to :menuselection:`Website app --> Contact Us`, then click :guilabel:`Edit` in the

Author: Because there are so many different website templates, I cannot guarantee that it will always be at the top-right

top-right of the screen to open the web editor. Click on the form to open the form configuration
settings. The following options are available to customize the contact form:

.. image:: opportunities_form/form-customization.png
   :align: center
   :alt: The form configuration settings on an Odoo website.

- :guilabel:`Action`: the default action for a contact form is :guilabel:`Send an Email`. Select
  :guilabel:`Create an Opportunity` from the drop-down list to capture the information in the *CRM*.
- :guilabel:`Sales Team`: choose sales team from the drop-down menu the opportunities from this form
  should be assigned to.
- :guilabel:`Salesperson`: if the opportunities should be assigned to a specific salesperson, select
  them from the drop-down menu. If no selection is made in this field, the opportunities are
  assigned based on the team's existing rules.

> - :guilabel:`Marked Fields`: use this field to alter how the form handles marked fields. The default

Reviewer: What is a "marked field"? Is that a required field? How do you make one required but one not? I might just link to a doc that explains how to do that, or add a little bit more info. I wasn't sure if it meant ALL the fields, etc.

  option is to treat marked fields as :guilabel:`Required`, which is the recommended setting.
- :guilabel:`Mark Text`: choose how :guilabel:`Marked Fields` should be identified. The default
  character is an asterisk (`*`).
- :guilabel:`Labels Width`: use this field to alter the pixel width of the labels, if desired.

> - :guilabel:`On Success`: choose how the webpage reacts after a customer successfully submits a

Reviewer: there's only 3 options, so you could explain them and what they do

  form.
- :guilabel:`URL`: if :guilabel:`Redirect` is selected in the :guilabel:`On Success` field, enter

>   the for the webpage where customers should be directed after successfully submitting a form.

Reviewer: ```suggestion
  the URL for the webpage where customers should be directed after successfully submitting a form.
```

- :guilabel:`Visibility`: use the drop-down menu to add any visibility conditions for this field, if

>   desired.

Reviewer: instead of listing all the options - if there is a doc already in the website app section that explains all of tihs, you could link to the various anchors or docs (if available)

- :guilabel:`Type`: choose a custom field option or an existing field type.
- :guilabel:`Input Type`: determine the type of information customers should input. Available
  options are :guilabel:`Text`, :guilabel:`Email`, :guilabel:`Telephone`, or :guilabel:`Url`. The
  selection made in this field limits the format that customers can use when entering information.
- :guilabel:`Label`: enter the name for the field.
- :guilabel:`Position`: choose the way the label is aligned with the rest of the form. The label can
  be hidden, above the field, to the left of the field, or right adjusted and closer to the field.
- :guilabel:`Description`: slide the toggle to add a description for the field, which can provide
  additional instructions to customers. Click under the field on the form to add the description.
- :guilabel:`Placeholder`: enter an example to guide users how to input information where
  formatting is important, such as a phone number or email address.
- :guilabel:`Default Value`: enter a value to include in the form by default if the customer does
  not provide information in the field. *It is not recommended to include a default value for
  required fields*.

> - :guilabel:`Required`: slide the toggle to mark this field as required if it **must** be filled in

Reviewer: I see! This is where the "marked" part I questioned earlier exists =D You could link to here if you wanted.

  for every submission.

> - :guilabel:`Visibility`: select when this field should be visible.

Reviewer: Since there's configurations for "Visible only if' you could add what that does and what needs to be filled in (perhaps an example of why you'd use this for creating opportunities).

- :guilabel:`Animation`: select if this field should have any animation.

.. image:: opportunities_form/field-customization.png
   :align: center
   :alt: The field configuration settings on an Odoo website.

View opportunities
==================

After a customer submits a contact form and an opportunity is created, it is assigned based on the
:ref:`form settings <crm/customize-contact-form>`. To view opportunities, navigate to
:menuselection:`'CRM app' --> Sales --> My Pipeline`.

.. note::

>    If leads are activated on the database, contact form submissions are generated as leads, not

Reviewer: is the default Opportunities, and they must activat leads? Just curious, ecause that coudl be adde-d I honestly am not sure!

Author: Opportunities are the default, leads need to be activated. I will link to the admonition block where this is called out

   opportunities. Navigate to :menuselection:`CRM app --> Leads` to view the newly created leads.

> On the :guilabel:`My Pipeline` dashboard, click on the Kanban card for an opportunity to open the
> details page. The information submitted by the customer is visible on the opportunity record.

Reviewer: ```suggestion
On the :guilabel:`My Pipeline` dashboard, click on an opportunity 
 card in the Kanban view to open the
details page. The information submitted by the customer is visible on the opportunity record.
```

.. note::
   As the contact form fields are customizable, the fields on the opportunity record where the form
   information is stored varies.

   If the preconfigured contact form is used, the :guilabel:`Subject` field is added to the
   :guilabel:`Title` field, and the content in the :guilabel:`Notes` field, which is labeled as

>    :guilabel:`Your Question`, is added to the :guilabel:`Internal Notes` tab.

Reviewer: If Subject is on the contact form, and is not visible on this screen, I believe the format is to use asterisks. I just had this issue ocme up with Sam when I references a field on a different page form where the user was currently on. It could also be reworded like this:

 If the preconfigured contact form is used, what was entered in the 
 *Subject* field populates the :guilabel:`Title` field,....

> .. seealso::

Reviewer: could add a website doc perhaps?

============================================
Create opportunities from a web contact form
============================================

Adding a contact form to a website makes it easy to convert visitors into leads and opportunities.
After a visitor submits their information, an opportunity can be created automatically, and assigned

> to a designated sales team.

Reviewer 2: ... (or salesperson)... is that correct?

.. _crm/customize-contact-form:

Customize contact forms
=======================

By default, the *Contact Us* page on an Odoo website displays a preconfigured contact form. This
form can be customized, as needed, to suit the needs of a specific sales team.

Navigate to :menuselection:`Website app --> Contact Us`, then click :guilabel:`Edit` in the

> top-right of the screen to open the web editor. Click on the form to open the form configuration

Reviewer 2: ```suggestion
top-right of the screen to open the web editor. Click on the form building block in the body of the webpage to open the form configuration
```

> settings. The following options are available to customize the contact form:

Reviewer 2: ```suggestion
settings on the right sidebar. The following options are available to customize the contact form from the :guilabel:`From` section of the right sidebar:
```

.. image:: opportunities_form/form-customization.png
   :align: center
   :alt: The form configuration settings on an Odoo website.

- :guilabel:`Action`: the default action for a contact form is :guilabel:`Send an Email`. Select

>   :guilabel:`Create an Opportunity` from the drop-down list to capture the information in the *CRM*.

Reviewer 2: ```suggestion
  :guilabel:`Create an Opportunity` from the drop-down list to capture the information in the *CRM* app.
```

> - :guilabel:`Sales Team`: choose sales team from the drop-down menu the opportunities from this form
>   should be assigned to.

Reviewer 2: ```suggestion
- :guilabel:`Sales Team`: choose a sales team from the drop-down menu that the opportunities from this form
  should be assigned to. This field **only** appears if the :guilabel:`Action` field is set to :guilabel:`Create an Opportunity`.
```

- :guilabel:`Salesperson`: if the opportunities should be assigned to a specific salesperson, select
  them from the drop-down menu. If no selection is made in this field, the opportunities are
  assigned based on the team's existing rules.
- :guilabel:`Marked Fields`: use this field to alter how the form handles marked fields. The default
  option is to treat marked fields as :guilabel:`Required`, which is the recommended setting.
- :guilabel:`Mark Text`: choose how :guilabel:`Marked Fields` should be identified. The default
  character is an asterisk (`*`).
- :guilabel:`Labels Width`: use this field to alter the pixel width of the labels, if desired.
- :guilabel:`On Success`: select how the webpage reacts after a customer successfully submits a
  form. :guilabel:`Nothing` keeps the customer on the same screen, with the addition of a
  confirmation message that the form was submitted successfully. :guilabel:`Redirect` sends the

>   customer to a new webpage, based on the address in the field below. :guilabel:`Show Message`

Reviewer 2: ```suggestion
  customer to a new webpage, based on the address provided in the :guilabel:`URL` field below. :guilabel:`Show Message`
```

  replaces the form with a preconfigured message that informs the customer someone should respond to
  them as soon as possible.
- :guilabel:`URL`: if :guilabel:`Redirect` is selected in the :guilabel:`On Success` field, enter

>   the URL for the webpage where customers should be directed after successfully submitting a form.

Reviewer 2: ```suggestion
  the URL for the webpage, where customers should be directed after successfully submitting a form.
```

- :guilabel:`Visibility`: use the drop-down menu to add any visibility conditions for this field, if
  desired.

.. image:: opportunities_form/form-submitted-successfully.png
   :align: center
   :alt: The screen displayed after a customer successfully submits a contact form.

.. important::

>    If *leads* are activated in CRM settings, selecting :guilabel:`Create an Opportunity` generates a

Reviewer 2: ```suggestion
   If *leads* are activated in *CRM* settings, selecting :guilabel:`Create an Opportunity` generates a
```

>    lead instead. To activate leads, navigate to :menuselection:`CRM app --> Configuration -->

Reviewer 2: ```suggestion
   lead, instead. To activate leads, navigate to :menuselection:`CRM app --> Configuration -->
```

>    Settings` and tick the :guilabel:`Leads` checkbox, then click :guilabel:`Save`.

Reviewer 2: ```suggestion
   Settings`, and tick the :guilabel:`Leads` checkbox. Then, click :guilabel:`Save`.
```

Customize contact form fields
-----------------------------

> In addition to the settings for the form, the settings for each field can be customized as well.

Reviewer 2: ```suggestion
In addition to the settings for the form, the settings for each field can be customized, as well.
```

> With the web editor menu still open, click into a field to open the field configuration settings.

Reviewer 2: ```suggestion
With the web editor menu still open, click into a field to open the :guilabel:`Field` configuration settings section on the sidebar.
```

The following options are available to customize a field:

- :guilabel:`Type`: choose a custom field option or an existing field type.
- :guilabel:`Input Type`: determine the type of information customers should input. Available
  options are :guilabel:`Text`, :guilabel:`Email`, :guilabel:`Telephone`, or :guilabel:`Url`. The
  selection made in this field limits the format that customers can use when entering information.
- :guilabel:`Label`: enter the name for the field.
- :guilabel:`Position`: choose the way the label is aligned with the rest of the form. The label can

>   be hidden, above the field, to the left of the field, or right adjusted and closer to the field.

Reviewer 2: do you want to consider tossing some FA icons in here for the corresponding options? Like, eye-slash for hidden, etc.? Might be cool. Just a suggestion, though. Totally up to you 👍

Author: I looked into this, but only the eye is an fa icon (the three arrows are svgs, not icons). Only including the one seemed out of place, I'd rather leave them all out for now

- :guilabel:`Description`: slide the toggle to add a description for the field, which can provide
  additional instructions to customers. Click under the field on the form to add the description.

> - :guilabel:`Placeholder`: enter an example to guide users how to input information where

Reviewer 2: ```suggestion
- :guilabel:`Placeholder`: enter an example to help users know how to input information where
```

  formatting is important, such as a phone number or email address.

> - :guilabel:`Default Value`: enter a value to include in the form by default if the customer does

Reviewer 2: ```suggestion
- :guilabel:`Default Value`: enter a value to include in the form, by default, if the customer does
```

  not provide information in the field. *It is not recommended to include a default value for
  required fields*.
- :guilabel:`Required`: slide the toggle to mark this field as required if it **must** be filled in
  for every submission.

> - :guilabel:`Visibility`: select when this field should be visible.

Reviewer 2: have you thought about adding any brief explanation for the second 'Visibility' field that appears beneath the first one,  wherein a Desktop and Mobile option (via icons) appears?

Author: same as above, these aren't icons, they are svgs, and there aren't any corresponding fa icons I can add. I will add more about the visibility options to clarify what is available :+1:

- :guilabel:`Animation`: select if this field should have any animation.

.. image:: opportunities_form/field-customization.png
   :align: center
   :alt: The field configuration settings on an Odoo website.

View opportunities
==================

> After a customer submits a contact form and an opportunity is created, it is assigned based on the

Reviewer 2: ```suggestion
After a customer submits a contact form, and an opportunity is created, it is assigned based on the
```

:ref:`form settings <crm/customize-contact-form>`. To view opportunities, navigate to
:menuselection:`'CRM app' --> Sales --> My Pipeline`.

.. note::
   If leads are activated on the database, contact form submissions are generated as leads, not
   opportunities. To activate leads, navigate to :menuselection:`CRM app --> Configuration -->

>    Settings` and tick the :guilabel:`Leads` checkbox, then click :guilabel:`Save`.

Reviewer 2: ```suggestion
   Settings`, and tick the :guilabel:`Leads` checkbox. Then, click :guilabel:`Save`.
```

>    Navigate to :menuselection:`CRM app --> Leads` to view the newly created leads.

Reviewer 2: ```suggestion
   Navigate to :menuselection:`CRM app --> Leads` to view the newly-created leads.
```

On the :guilabel:`My Pipeline` dashboard, click on an opportunity card in the Kanban view to open

> the details page. The information submitted by the customer is visible on the opportunity record.

Reviewer 2: ```suggestion
the opportunity record. The information submitted by the customer is visible on the opportunity record.
```

.. note::

>    As the contact form fields are customizable, the fields on the opportunity record where the form
>    information is stored varies.

Reviewer 2: ```suggestion
   As the contact form fields are customizable, the fields on the opportunity record, where the form
   information is stored, varies accordingly.
```

  option is to treat marked fields as :guilabel:`Required`, which is the recommended setting.
- :guilabel:`Mark Text`: choose how :guilabel:`Marked Fields` should be identified. The default
  character is an asterisk (`*`).
- :guilabel:`Labels Width`: use this field to alter the pixel width of the labels, if desired.
- :guilabel:`On Success`: select how the webpage reacts after a customer successfully submits a
  form. :guilabel:`Nothing` keeps the customer on the same screen, with the addition of a
  confirmation message that the form was submitted successfully. :guilabel:`Redirect` sends the
  customer to a new webpage, based on the address provided in the :guilabel:`URL` field below.
  :guilabel:`Show Message` replaces the form with a preconfigured message that informs the customer
  someone should respond to them as soon as possible.
- :guilabel:`URL`: if :guilabel:`Redirect` is selected in the :guilabel:`On Success` field, enter
  the URL for the webpage, where customers should be directed after successfully submitting a form.
- :guilabel:`Visibility`: use the drop-down menu to add any visibility conditions for this field, if
  desired.

> .. image:: opportunities_form/form-submitted-successfully.png
>    :align: center
>    :alt: The screen displayed after a customer successfully submits a contact form.
> 

Reviewer 3: See my previous point :)
```suggestion
```

- :guilabel:`Placeholder`: enter an example to help users know how to input information where
  formatting is important, such as a phone number or email address.
- :guilabel:`Default Value`: enter a value to include in the form, by default, if the customer does
  not provide information in the field. *It is not recommended to include a default value for
  required fields*.
- :guilabel:`Required`: slide the toggle to mark this field as required if it **must** be filled in
  for every submission.
- :guilabel:`Visibility`: select when this field should be visible. Use the button on the left to
  choose whether to show or hide this field on a desktop users. Use the button on the right to
  choose whether to show or hide this field to mobile users.
- :guilabel:`Animation`: select if this field should have any animation.

.. image:: opportunities_form/field-customization.png
   :align: center
   :alt: The field configuration settings on an Odoo website.

View opportunities
==================

After a customer submits a contact form, and an opportunity is created, it is assigned based on the
:ref:`form settings <crm/customize-contact-form>`. To view opportunities, navigate to

> :menuselection:`'CRM app' --> Sales --> My Pipeline`.

Reviewer 3: Not sure why single quotes are around _CRM app_, can remove :)
```suggestion
:menuselection:`CRM app --> Sales --> My Pipeline`.
```
